src/store/modules/product/actions.ts

```diff
@@ -83,10 +83,18 @@ const actions: ActionTree<ProductState, RootState> = {
     commit(types.PRODUCT_ADD_TO_UPLD_PRDTS, { product: payload })
   },
 
-  updateCurrentProduct ({ commit, state }, payload) {
+  async updateCurrentProduct ({ commit, state }, payload) {
     // search in uploadProducts that if the clicked product is already in the upload list and set it as current product
-    const currentProduct = state.uploadProducts[payload.product.sku]
-    commit(types.PRODUCT_CURRENT_UPDATED, { product: currentProduct ? currentProduct : payload.product })
+    const currentProduct = state.uploadProducts[payload]
+    if(currentProduct) {
+      commit(types.PRODUCT_CURRENT_UPDATED, { product: currentProduct })
+    } else {
+      const resp = await ProductService.fetchProducts({
```

Comment:
Handle the case when API returns an error.

Reply:
Sir, updated code.